### PR TITLE
fix: store and actions issues

### LINKED ISSUE #1869: State modal shouldn't open with empty edit
[LOOM DEMO](https://www.loom.com/share/7dbd19d7e7854a82a870c35c12f3b00c)

### PR DESCRIPTION
**Pull request by assimhabeek**
## Related Issue(s)

Fixes #1869

### CODE REVIEW

File: libs/frontend/modules/store/src/store/apis/api.utils.ts [hunk 1]
```diff
@@ -65,6 +65,9 @@ export const makeActionCreateInput = (
             ApiAction: {
               connect: { where: { node: { id: action.errorActionId } } },
```
Comment (webberwang) [inline libs/frontend/modules/store/src/store/apis/api.utils.ts:66]: we can use `ApiAction: connectNode(action.errorActionId)` helper

File: libs/frontend/modules/store/src/store/models/store.model.ts [hunk 2]
```diff
@@ -34,17 +30,31 @@ export class Store
...
   @modelAction
-  writeCache({ id, name, actions, api }: IStoreDTO) {
+  updateState(data: IPropData) {
+    this._state = merge(this._state, data)
+  }
+
+  @computed
+  get state() {
+    const state: IPropData = merge(
+      { ...this._api.defaults },
```
Comment (webberwang) [inline libs/frontend/modules/store/src/store/models/store.model.ts:46]: don't have to spread here right?

File: libs/frontend/modules/store/src/store/models/store.model.ts [hunk 3]
```diff
@@ -58,28 +68,21 @@ export class Store
...
-      this._api.defaults,
-      this._storeActions,
-    )
-
-    for (const key of keys(this._storeActions)) {
-      state[key].run = createActionFn(state[key].action, state)
-    }
-
-    return state
+  get _actionsRunners() {
```
Comment (webberwang) [inline libs/frontend/modules/store/src/store/models/store.model.ts:78]: I think making private is good here, and we could remove from the `IStore` interface, this way the interfaces are only public methods which simplifies it. same for all classes when applicable

**Review by webberwang — CHANGES_REQUESTED** [review]

File: libs/frontend/modules/store/src/store/models/store.model.ts (hunk 2 above)
Comment (assimhabeek) [reply libs/frontend/modules/store/src/store/models/store.model.ts:46]: we do have to spread to clone or else the error will occur 
[image](https://user-images.githubusercontent.com/33001357/193041235-50bb4fba-d371-4753-81d2-eb0ff2ed8499.png)

**Pull request merged**